### ISSUE TITLE
Refactor `PlatformState` into a singleton

### ISSUE THREAD

**Issue by imalygin** [hashgraph/hedera-services#11771]
Currently, `PlatformState` is an instance of `MerkleLeaf` and it deviates from the idea of having only three types in the Merkle tree (KV storage, queue, and singleton). It should be refactored to a singleton.

When this ticket is complete, `PlatformState` will be migrated to the protobuf format and accessed through the State API only.

`PlatformState` contains several fields to be deleted:

`epochHash` - we no longer support emergency reconnect
`nextEpochHash` - same
`uptimeData` - currently not utilized for anything. This is a part of a feature that was not fully implemented, and it's implementation should be reconsidered separately

The unit tests associated with this change are NOT part of this ticket. This ticket will NOT result in a mergable PR. Criteria of Done for this ticket is getting informal signoffs on the draft PR so that unit tests can be updated (another ticket) without rework.

**Comment by imalygin** [comment]
Blocked by #13084

**Comment by imalygin** [comment]
Blocked by #14220 
Blocked by #14230

**Comment by imalygin** [comment]
Blocked by https://github.com/hashgraph/hedera-services/issues/11773

**Comment by imalygin** [comment]
It makes sense to implement https://github.com/hashgraph/hedera-services/issues/11773 because it will be much easier to migrate from `PlatformState` to a singleton once it's done. 

For example, there is `buildGenesisState` method:

```
    public static ReservedSignedState buildGenesisState(
            @NonNull final PlatformContext platformContext,
            @NonNull final AddressBook addressBook,
            @NonNull final SoftwareVersion appVersion,
            @NonNull final MerkleRoot stateRoot) 
```

This is where we create initial platform state. Currently, `MerkleRoot` is implemented by two classes - `State` (current merkle root) and `MerkleHederaState` (to become a merkle root once #11773 is complete). If `MerkleHederaState` was a root, the migration would be more natural as `MerkleHederaState` implements `HederaState` interface which is a main interface of State API

**Comment by imalygin** [comment]
Blocked by #11773
